src/renderer/components/resource-metrics/resource-metrics-text.tsx

```diff
@@ -4,22 +4,38 @@
  */
 
 import React from "react";
-import type { PodMetricData } from "../../../common/k8s-api/endpoints";
+import type { MetricData } from "../../../common/k8s-api/endpoints/metrics.api";
 import { getMetricLastPoints } from "../../../common/k8s-api/endpoints/metrics.api";
 import { bytesToUnits } from "../../utils";
 import { Badge } from "../badge";
 import { DrawerItem } from "../drawer";
 
+export interface ResourceMetricsTextMetrics {
+  cpuUsage?: MetricData;
+  cpuRequests?: MetricData;
+  cpuLimits?: MetricData;
+  memoryUsage?: MetricData;
+  memoryRequests?: MetricData;
+  memoryLimits?: MetricData;
+}
+
 export interface ResourceMetricsTextProps {
-  metrics: PodMetricData | null | undefined;
+  metrics: ResourceMetricsTextMetrics | null | undefined;
 }
 
 export function ResourceMetricsText({ metrics }: ResourceMetricsTextProps) {
   if (!metrics) {
     return null;
   }
 
-  const { cpuUsage, cpuRequests, cpuLimits, memoryUsage, memoryRequests, memoryLimits } = getMetricLastPoints(metrics);
+  const {
```

Comment:
Are the added default values here to fix a bug or something? :)

Reply:
Slightly, the new and better return type of `getMetricLastPoints` is 

```
function getMetricLastPoints<Keys extends string>(metrics: Partial<Record<Keys, MetricData>>): Partial<Record<Keys, number>>
```

which correctly encodes that some of the metrics might not have last values. Typescript then complains that `number | undefined` shouldn't be used with the `>` operator.

Reply:
Ah. Maybe the commit could communicate something like this in the future? Resolved :)